plugin/src/withDateTimePickerStyles.ts

```diff
@@ -0,0 +1,258 @@
+    if (android.datePicker?.light) {
+      config.modResults = setAndroidColors(
+        config.modResults,
+        android.datePicker.light,
+        DATE_PICKER_ALLOWED_ATTRIBUTES,
+      );
+    }
+
+    if (android.timePicker?.light) {
+      config.modResults = setAndroidColors(
+        config.modResults,
+        android.timePicker.light,
+        TIME_PICKER_ALLOWED_ATTRIBUTES,
+      );
+    }
+
+    return config;
+  });
+
+  newConfig = withAndroidColorsNight(newConfig, (config) => {
+    if (android.datePicker?.dark) {
+      config.modResults = setAndroidColors(
+        config.modResults,
+        android.datePicker.dark,
+        DATE_PICKER_ALLOWED_ATTRIBUTES,
+      );
+    }
+
+    if (android.timePicker?.dark) {
+      config.modResults = setAndroidColors(
+        config.modResults,
+        android.timePicker.dark,
+        TIME_PICKER_ALLOWED_ATTRIBUTES,
+      );
+    }
+
+    return config;
+  });
+
+  newConfig = withAndroidStyles(newConfig, (config) => {
+    if (android.datePicker) {
+      config.modResults = setAndroidDatePickerStyles(
+        config.modResults,
+        android.datePicker,
+      );
+    }
+
+    if (android.timePicker) {
+      config.modResults = setAndroidTimePickerStyles(
+        config.modResults,
+        android.timePicker,
+      );
+    }
+
+    return config;
+  });
+
+  return newConfig;
+};
+
+const setAndroidDatePickerStyles = (
+  styles: ResourceXML,
+  {light, dark}: DatePickerTheme,
+): ResourceXML => {
+  styles = Object.keys({...light, ...dark}).reduce((acc, attr) => {
+    const {attrName, colorName} =
+      DATE_PICKER_ALLOWED_ATTRIBUTES[attr as keyof DatePickerProps];
+
+    return assignStylesValue(acc, {
+      add: true,
+      parent: {
+        name: DATE_PICKER_STYLE_NAME,
+        parent: 'Theme.AppCompat.Light.Dialog',
+      },
+      name: attrName,
+      value: `@color/${colorName}`,
+    });
+  }, styles);
+
+  styles = assignStylesValue(styles, {
+    add: true,
+    parent: getAppThemeLightNoActionBarGroup(),
+    name: DATE_PICKER_THEME_ATTRIBUTE,
+    value: `@style/${DATE_PICKER_STYLE_NAME}`,
+  });
+
+  return styles;
+};
+
+const setAndroidTimePickerStyles = (
+  styles: ResourceXML,
+  {light, dark}: TimePickerTheme,
+): ResourceXML => {
+  styles = Object.keys({...light, ...dark}).reduce((acc, attr) => {
+    const {attrName, colorName} =
+      TIME_PICKER_ALLOWED_ATTRIBUTES[attr as keyof TimePickerProps];
+
+    return assignStylesValue(acc, {
+      add: true,
+      parent: {
+        name: TIME_PICKER_STYLE_NAME,
+        parent: 'android:Widget.Material.Light.TimePicker',
+      },
+      name: attrName,
+      value: `@color/${colorName}`,
+    });
+  }, styles);
+
+  styles = assignStylesValue(styles, {
+    add: true,
+    parent: getAppThemeLightNoActionBarGroup(),
+    name: TIME_PICKER_THEME_ATTRIBUTE,
+    value: `@style/${TIME_PICKER_STYLE_NAME}`,
+  });
+
+  return styles;
+};
```

Comment:
Can we have a generic method to dedupe these two functions? They're doing the same thing with different arguments.